assets/chef-recipes/cookbooks/docker/recipes/default.rb

```diff
@@ -54,8 +54,18 @@
     # The current version is always installed
     command 'sudo zypper install -y docker'
   end
-  execute 'Enable and start Docker service' do
+  execute 'Enable Docker service' do
     command 'sudo systemctl enable docker'
+  end
+  execute 'Start Docker service' do
     command 'sudo systemctl start docker'
   end
 end
+
+user = ENV['SUDO_USER']
+home_dir = Dir.home(user)
+gnupg_dir = File.join(home_dir, '.gnupg')
+execute 'Chown ~.gnupg/ directory to sudo user' do
+  command "sudo chown -R #{user}:#{user} #{gnupg_dir}"
```

Comment:
There might not be a group with the same name as the user. You should instead use the `chown -R user:` form. This way the default group for the current user will be used when changing the ownership of the file.